torch/jit/__init__.py

```diff
@@ -957,6 +995,13 @@ def _qualified_name(obj):
     return module_name + "." + name
 
 
+def _recursive_script(fn):
```

Comment:
small nit: the eventual API will just look like
```
@torch.jit.script
def foo(...)
```

So it would be nicer if we just did this in our tests for now:

```
with self.enableRecursiveScript():
   @torch.jit.script
   def foo(...)
      ..
```

Otherwise we are going to have to introduce similar things for script_method, where it will require additional threading of state since the creation of script_method and when it actually compiles is different.